internal/database/models/auth.go

```diff
@@ -5,14 +5,17 @@ import (
 	"gorm.io/gorm"
 )
 
+// User stores all users that have ever signed in with a
+// uwblueprint or theshoeproject domains (stored in Hd field)
 type User struct {
 	gorm.Model
-	Username string `json:"username" gorm:"unique; not null"`
-	Password string `json:"password" gorm:"not null"`
+	Email string `json:"email" gorm:"unique; not null"`
+	Hd    string `json:"hd" gorm:"not null"`
```

Comment:
Hmm I'm wondering if we can write a CHECK constraint here to limit the Hd values to one of the domains, but idk how hard it is with gorm.

Reply:
see above